### ISSUE TITLE
Importing a partial seems to make filenames incorrect

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-stache#544]
I am seeing the following error in the console:

```
can-log.js:98 components/page-dashboard/client-projects/client-projects-os-projects-modal.stache:39: Unable to find key "toFixed()".
```

However, `toFixed()` is actually used in 

`components/page-dashboard/client-projects/client-projects.stache:39` (actually line 41)

It seems the path is wrong because `client-projects.stache` imports a template like:

```
<can-import from="./client-projects-os-projects-modal.stache" value:to="scope.vars.osProjectsModal" />
	{{ >scope.vars.osProjectsModal }}
```

Removing the `<import>` and `{{ >scope.vars.osProjectsModal }}` makes the error correctly report.

**Comment by phillipskevin** [comment]
The fileName is set on the templateContext when the partial is rendered. So anything that happens asynchronously will end up seeing the filename of the partial.

It's reproducable by this simple template:

```
        {{#if foo}}
            {{ aFunction() }}
        {{/if}}

        <can-import
            from="partial.stache"
            value:to="scope.vars.partial" />

        {{ >scope.vars.partial }}
```

If you set `foo` after the partial is rendered and `aFunction` doesn't exist, you get the wrong filename in the warning.

I'll think about how we can fix this.

**Comment by justinbmeyer** [comment]
Could we make it a getter so it doesn’t need to be set?

**Comment by phillipskevin** [comment]
What would the getter do?

**Comment by phillipskevin** [comment]
closed by #547.
